### DISCUSSION TITLE
classes/class_scrollcontainer

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#11] [User-contributed notes]
# classes/class_scrollcontainer

Inherits: Container< Control< CanvasItem< Node< Object Inherited By: EditorInspector A container used to provide scrollbars to a child control when needed. Description: A container used to provide ...

https://docs.godotengine.org/en/latest/classes/class_scrollcontainer.html

**Comment by SpyrexDE** [comment]
# SmoothScroll Addon
For enhanced smooth scrolling functionality including inertia, "overdragging", and touch support there is an MIT-licensed addon available on GitHub:
https://github.com/SpyrexDE/SmoothScroll
> This add-on is actively maintained by myself and benefits greatly from open source contributors.

**Comment by stayinsidegames** [comment]
If you are using ScrollContainer to make a list of items, the [ItemContainer](https://docs.godotengine.org/en/stable/classes/class_itemlist.html) node will also add a scrollbar and allow scrolling, just like ScrollContainer.

**Comment by admin** [comment]
*Originally posted by [**GofusProject**](https://github.com/GofusProject) on 2026-06-10 16:42:58 UTC*

> For those who, like me, can't figure out how to edit the scroll bar thickness: it seems that this properties depends only on the increment icons (the arrows at each end of the scroll bar).

**Comment by Gerinse24** [comment]
Can we get more control over the width, thickness of scrollbars?

**Comment by Gerinse24** [comment]
Here's how to control the width of your scrollbar:

1. Select your ScrollContainer, in the inspector go down to Theme and Add a new Theme. 
2. Click Manage Items, and add VScrollBar
3. Add a StyleBoxItem for grabber and scroll
4. Choose StyleBoxFalt for both grabber and scroll
5. Under Content Margins, set Left and Right to 0 to hide it, or chose your desired value. Do this for both of them.
6. Follow the same process for horizontal scroll bars but set the Top and Bottom Content Margins.
7. Hate life a little less.

**Reply by Gerinse24** [reply]
This applies to pretty much any node that has a scrollbar, you just select the node that has it (RichTextLabel, ScrollContainer etc) and follow the steps.
